cli/azd/pkg/async/task.go

```diff
@@ -0,0 +1,149 @@
+package async
+
+// Task represents a long running async operation
+type Task[R comparable] struct {
+	isComplete    bool
+	hasResult     bool
+	result        R
+	resultChannel chan R
+	error         error
+}
+
+// Runs the specified taskFn as a go routine
+func (t *Task[R]) Run(taskFn TaskRunFunc[R]) {
+	go func() {
+		context := NewTaskContext(t)
+
+		taskFn(context)
+		t.complete(context.result, context.error)
```

Comment:
What if you just pass the `context` and let `complete()` to get result and error.

Reply:
Since there a couple Task variations the complete method would not be able to take into of a common TaskContext.  Opting for explicitly passing in the result & error respectively.